src/ORM/ResultSet.php

```diff
@@ -524,8 +525,26 @@ protected function _groupResult($row)
             }
         }
 
+        // In PostgreSQL, identifiers — table names, column names, constraint names, etc. — are limited to a
+        // maximum length of 63 bytes. Identifiers longer than 63 characters can be used, but they will be truncated
+        // to the allowed length of 63.
+        $sanitize_alias_length = function ($a, $b) {
+            if ($this->_driver instanceof Postgres) {
```

Comment:
It would probably be better to use the driver check to conditionally apply `array_intersect_ukey()`, so that the custom function is only applied for Postgres.

But I'm not sure if this is really appreciated by Postgres users in the first place, as the number of calls to `$sanitize_alias_length()` can easily go into the tens of thousands with larger results (10 fields with 10 matching keys results in 32 calls per row, and every additional field (for example associations) means 7 additional calls per row), that's quite some overhead that could easily be avoided by simply sticking to the Postgres limitations.

Reply:
I've added a new approach avoiding your concerns. thanks pointing this out!

Primarly stumpled upon this when converting a cakephp application from mariadb to postgresql.

How are you handling table and column names in case they get to long vs. cakephp naming conventions?

Reply:
to further limit the performance impact we can modify in `protected function _calculateColumnMap($query)` directly ?

Reply:
> How are you handling table and column names in case they get to long vs. cakephp naming conventions?

Personally I'd just break with the conventions in that case, and provide the required/limited aliases/identifiers manually.

> to further limit the performance impact we can modify in `protected function _calculateColumnMap($query)` directly ?

I'm not sure if the shortening will work out in the first place, you'd need to add some tests, but I could imagine that this won't help when for example foreign keys don't match.

I'm also not sure if users would really want this behavior instead of being forced to configure the application and the schema accordingly. Personally I wouldn't feel overly confident leaving an application in such a fragile state.

Reply:
why is it fragile? it's default postres behaviour and perfectly fine as long as identifiers are still unique. (see my comment below).

works fine and is not affecting foreign key matches or anything else. would be willing to add tests if appreciated.

Reply:
I'd call it "fragile" because the application is configured with keys that practically do not exist, IMHO that's not an overly reliable construct, there's so many places where things could go wrong.

Foreign keys were just one example, there's lots of other places where things might break, practically everywhere where conventions are used or the full hardcoded names. Forms for example, where the field names wouldn't match the identifiers in the entities. Or patching, where the marshaller will have shortened identifiers in the entities, and non-shortened identifiers in the request data. Or application rules that run tests against non-shortened identifiers.

Reply:
The property names of entities are not affected by this change.

Reply:
I know, I probably expressed myself a little vaguely, I didn't mean to say that it's your patch that will break these things, I mean that the shortening by Postgres can break these things, and that  this patch won't prevent that. While it circumvents the error in the result set, it will leave the application in this "fragile" state where lots of things can go wrong.